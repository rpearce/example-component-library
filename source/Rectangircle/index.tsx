import React, { FC } from 'react'
import Circle from '../Circle'
import Rectangle from '../Rectangle'

const Rectangircle: FC = () => (
  <>
    <Circle
      className="rl-rectangircle__circle"
      desc="A blue circle"
      fill="#7ed6df"
      size={200}
      title="Water planet"
    />
    <Rectangle
      className="rl-rectangircle__rect"
      desc="A rectangle that is 4 times taller than it is wide"
      fill="#30336b"
      height={100}
      title="Minimalist building"
      width={400}
    />
  </>
)

export default Rectangircle
