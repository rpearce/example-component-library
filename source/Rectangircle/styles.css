.rl-rectangircle__circle { margin: 1em; }
.rl-rectangircle__rect   { margin: 1em; }
