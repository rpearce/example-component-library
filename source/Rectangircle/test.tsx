import React from 'react'
import { render } from '@testing-library/react'
import Rectangircle from './index'

test('with no props', () => {
  const { asFragment } = render(<Rectangircle />)

  expect(asFragment()).toMatchSnapshot()
})
