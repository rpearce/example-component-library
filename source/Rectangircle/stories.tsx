import React from 'react'
import { storiesOf } from '@storybook/react'
import { withA11y } from '@storybook/addon-a11y'
import { Rectangircle } from '../../dist/esm'
import '../../.storybook/base.css'
import '../../dist/css/Rectangircle/styles.css'

const stories = storiesOf('Rectangircle', module)

stories.addDecorator(withA11y)

stories.add('default', () => <Rectangircle />)
