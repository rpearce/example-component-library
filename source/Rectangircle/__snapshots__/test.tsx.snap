// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with no props 1`] = `
<DocumentFragment>
  <svg
    class="rl-rectangircle__circle"
    fill="#7ed6df"
    height="200"
    role="img"
    viewBox="0 0 100 100"
    width="200"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Water planet
    </title>
    <desc>
      A blue circle
    </desc>
    <circle
      cx="50"
      cy="50"
      r="50"
    />
  </svg>
  <svg
    class="rl-rectangircle__rect"
    fill="#30336b"
    height="100"
    role="img"
    viewBox="0 0 400 100"
    width="400"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Minimalist building
    </title>
    <desc>
      A rectangle that is 4 times taller than it is wide
    </desc>
    <rect
      height="100"
      width="400"
    />
  </svg>
</DocumentFragment>
`;
